Very long vulnerability descriptions cause `xelatex` failure during report generation

# 🐛 Summary #

If a CyHy customer report includes a finding with a very long description, the report will fail to generate.

## To reproduce ##

Attempt to generate a CyHy report for an entity with an open ticket for a finding with a very long description (e.g. [this one](https://www.tenable.com/plugins/nessus/194919), whose description is currently 51,654 bytes): `cyhy-report --cyhy-section cyhy --scan-section scan ENTITY`

Note failure output:
```
Traceback (most recent call last):
  File "/usr/local/bin/cyhy-report", line 11, in <module>
    generate_report.main()
  File "/usr/local/lib/python2.7/dist-packages/cyhy_report/customer/generate_report.py", line 3643, in main
    was_encrypted, results = generator.generate_report()
  File "/usr/local/lib/python2.7/dist-packages/cyhy_report/customer/generate_report.py", line 348, in generate_report
    self.__generate_final_pdf()
  File "/usr/local/lib/python2.7/dist-packages/cyhy_report/customer/generate_report.py", line 3551, in __generate_final_pdf
    assert return_code == 0, "xelatex pass 1 of 3 return code was %s" % return_code
AssertionError: xelatex pass 1 of 3 return code was 1
```

## Expected behavior ##

The CyHy report PDF should be generated normally.

## Any helpful log output or screenshots ##

XeLaTex output after failure:

```console
Underfull \hbox (badness 1189) in paragraph at lines 2968--2968
[]|\TU/Arial(0)/m/n/10 Splunk Enterprise < 8.1.14,
 []

! TeX capacity exceeded, sorry [input stack size=5000].
\@makeother #1->\catcode `#1
                            12\relax
l.2972 ...ation{'}s self-reported version number.} 
                                                   \\
If you really absolutely need more capacity,
you can ask a wizard to enlarge me.


Here is how much of TeX's memory you used:
 41357 strings out of 494868
 792782 string characters out of 6175714
 1445703 words of memory out of 10000000
 44502 multiletter control sequences out of 15000+600000
 535069 words of font info for 117 fonts, out of 8000000 for 9000
 14 hyphenation exceptions out of 8191
 5000i,19n,8326p,64577b,676s stack positions out of 5000i,500n,10000p,30000000b,80000s
Output written on report.pdf (109 pages).
```

--- a/cyhy_report/customer/generate_report.py
+++ b/cyhy_report/customer/generate_report.py
@@ -3432,6 +3432,18 @@ class ReportGenerator(object):
             t["first_detected_time_tex"] = t["first_detected"].strftime("{%H}{%M}{%S}")
             t["last_detected_date_tex"] = t["last_detected"].strftime("{%d}{%m}{%Y}")
             t["last_detected_time_tex"] = t["last_detected"].strftime("{%H}{%M}{%S}")
+            # Trim long descriptions so that they don't overflow the page
+            # (LaTeX table cells cannot span pages, as far as I know) or cause
+            # other issues. For more information, see:
+            # - https://github.com/cisagov/cyhy-reports/issues/124
+            # - https://github.com/cisagov/cyhy-reports/issues/123
+            # NOTE: I wanted to make the cutoff 4096 characters, but that barely
+            # fit on a single page when there was a single affected host.  I
+            # went with 3072 instead to save space on the page for when there
+            # are many affected hosts listed.
+            if len(t["description"]) > 3072:
+                t["description"] = t["description"][:3072] + \
+                    "... (Truncated; the full description is available in the findings attachment.)"
 
         result["mitigations"] = self.__results["mitigations"]
         for t in result["mitigations"]:

--- a/cyhy_report/customer/generate_report.py
+++ b/cyhy_report/customer/generate_report.py
@@ -3440,8 +3440,8 @@ class ReportGenerator(object):
             # Trim long descriptions so that they don't overflow the page
             # (LaTeX table cells cannot span pages, as far as I know) or cause
             # other issues. For more information, see:
-            # - https://github.com/cisagov/cyhy-reports/issues/124
             # - https://github.com/cisagov/cyhy-reports/issues/123
+            # - https://github.com/cisagov/cyhy-reports/issues/124
             if len(t["description"]) > FINDING_DESCRIPTION_MAX_DISPLAY_LENGTH:
                 t["description"] = t["description"][
                     :FINDING_DESCRIPTION_MAX_DISPLAY_LENGTH] + \

--- a/cyhy_report/customer/generate_report.py
+++ b/cyhy_report/customer/generate_report.py
@@ -190,6 +190,11 @@ POTENTIAL_NMI_SERVICES = [
     "telnet",         # Telnet
 ]
 
+# I wanted to make the cutoff 4096 characters, but that barely fit on a single
+# page when there was a single affected host.  I went with 3072 instead to save
+# space on the page for when there are many affected hosts listed.
+FINDING_DESCRIPTION_MAX_DISPLAY_LENGTH = 3072
+
 def SafeDataFrame(data=None, *args, **kwargs):
     """A wrapper around pandas DataFrame so that empty lists still
     return a DataFrame with columns if requested."""
@@ -3437,12 +3442,9 @@ class ReportGenerator(object):
             # other issues. For more information, see:
             # - https://github.com/cisagov/cyhy-reports/issues/124
             # - https://github.com/cisagov/cyhy-reports/issues/123
-            # NOTE: I wanted to make the cutoff 4096 characters, but that barely
-            # fit on a single page when there was a single affected host.  I
-            # went with 3072 instead to save space on the page for when there
-            # are many affected hosts listed.
-            if len(t["description"]) > 3072:
-                t["description"] = t["description"][:3072] + \
+            if len(t["description"]) > FINDING_DESCRIPTION_MAX_DISPLAY_LENGTH:
+                t["description"] = t["description"][
+                    :FINDING_DESCRIPTION_MAX_DISPLAY_LENGTH] + \
                     "... (Truncated; the full description is available in the findings attachment.)"
 
         result["mitigations"] = self.__results["mitigations"]

